src/ROUTES.ts

```diff
@@ -448,7 +448,21 @@ const ROUTES = {
     },
     WORKSPACE_RATE_AND_UNIT: {
         route: 'workspace/:policyID/rateandunit',
-        getRoute: (policyID: string) => `workspace/${policyID}/rateandunit` as const,
+        getRoute: (policyID: string, unit?: string, rate?: string) =>
+            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+            `workspace/${policyID}/rateandunit${unit || rate ? '?' : ''}${unit ? `unit=${unit}` : ''}${unit && rate ? '&' : ''}${rate ? `rate=${rate}` : ''}` as const,
+    },
+    WORKSPACE_RATE_AND_UNIT_RATE: {
+        route: 'workspace/:policyID/rateandunit/rate',
+        getRoute: (policyID: string, unit?: string, rate?: string) =>
+            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+            `workspace/${policyID}/rateandunit/rate${unit || rate ? '?' : ''}${unit ? `unit=${unit}` : ''}${unit && rate ? '&' : ''}${rate ? `rate=${rate}` : ''}` as const,
+    },
+    WORKSPACE_RATE_AND_UNIT_UNIT: {
+        route: 'workspace/:policyID/rateandunit/unit',
+        getRoute: (policyID: string, unit?: string, rate?: string) =>
+            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+            `workspace/${policyID}/rateandunit/unit${unit || rate ? '?' : ''}${unit ? `unit=${unit}` : ''}${unit && rate ? '&' : ''}${rate ? `rate=${rate}` : ''}` as const,
```

Comment:
This entire block is confusing. Is there a way to simplify the parameters based on what the route actually ends with? This way, there is no need to conditionally check if it's rate or unit that is set.

For instance, for the `WORKSPACE_RATE_AND_UNIT_RATE` route, just return `workspace/${policyID}/rateandunit/rate`, and then load the rate value from Onyx.

Reply:
Am I allowed to create a new Onyx key for this?

Reply:
Adding new Onyx keys usually needs to be coordinated with the backend to ensure the new key is being populated properly. I might be missing some context for the rest of the question.

What Onyx key are you wanting to add and what does that have to do with the suggestion here (which I agree with) to clean up the routes?

Reply:
No it will be a temporary key. No need to sync it with backend.

Reply:
Please answer my question:

What Onyx key are you wanting to add and what does that have to do with the suggestion here?

Reply:
> What Onyx key are you wanting to add and what does that have to do with the suggestion here?

I haven't thought of a name yet, but the Onyx key will behave similar to Task key, whose function would be to hold the values temporarily while we are editing the RateUnit form.

Reply:
@akinwale Can you give another pass at the PR?